Tests/Grid/Mapping/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,221 @@
+        $value = 'bar';
+        $fieldMapping = [
+            $field => [
+                'type' => $value
+            ]
+        ];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+        $this->assertEquals($value, $this->metadata
+            ->getFieldMappingType($field));
+    }
+
+    public function testSetterGroupByWithString()
+    {
+        $groupBy = 'groupBy';
+
+        $this->metadata->setGroupBy($groupBy);
+
+        $this->assertAttributeEquals(
+            $groupBy,
+            'groupBy',
+            $this->metadata
+        );
+    }
+
+    public function testGetterGroupByWithString()
+    {
+        $groupBy = 'groupBy';
+
+        $this->metadata->setGroupBy($groupBy);
+        $this->assertEquals($groupBy, $this->metadata->getGroupBy());
+    }
+
+    public function testSetterNameWithString()
+    {
+        $name = 'name';
+
+        $this->metadata->setName($name);
+
+        $this->assertAttributeEquals(
+            $name,
+            'name',
+            $this->metadata
+        );
+    }
+
+    public function testGetterNameWithString()
+    {
+        $name = 'name';
+
+        $this->metadata->setName($name);
+
+        $this->assertEquals($name, $this->metadata->getName());
+    }
+
+    public function testGetColumnsFromMappingWithoutTypeReturnException()
+    {
+        $this->expectException(\Exception::class);
+
+        $field = 'foo';
+        $value = 'bar';
+        $fieldMapping = [
+            $field => [
+                'type' => $value
+            ]
+        ];
+
+        $columnsMock = $this->createMock(Columns::class);
+        $columnsMock->method('hasExtensionForColumnType')
+                    ->with($value)
+                    ->willReturn(false);
+
+        $this->metadata->setFields(['foo' => $field]);
+        $this->metadata->setFieldsMappings($fieldMapping);
+        $this->metadata->getColumnsFromMapping($columnsMock);
+    }
+
+    public function testGetColumnsFromMapping()
+    {
+        $field = 'foo';
+        $field2 = 'foo2';
+        $value = 'bar';
+        $value2 = 'bar';
+        $fieldMapping = [
+            $field => [
+                'type' => $value
+            ],
+            $field2 => [
+                'type' => $value2
+            ]
+        ];
+
+        $columnsMockClone = $this->getMockForAbstractClass(Column::class);
+
+        $columnsMock = $this->createMock(Columns::class);
+        $columnsMock->method('hasExtensionForColumnType')
+                    ->with($value)
+                    ->willReturn(true);
+
+        $columnsMock->method('getExtensionForColumnType')
+                    ->with($value)
+                    ->willReturn($columnsMockClone);
+
+        $this->metadata->setFields(['foo' => $field]);
+        $this->metadata->setFieldsMappings($fieldMapping);
+        $columns = $this->metadata->getColumnsFromMapping($columnsMock);
+
+        $this->assertInstanceOf('\SplObjectStorage', $columns);
+    }
+}
```

Comment:
One last note: you should remove suffixes from test names if there are no other test cases. For example
`testGetterNameWithString`: here `withString` is totally useless (and even deceptive) because I will looking for a different case (like `withArray` or `withInt` or whatever) and when I don't see it, I'm forced to go into the code and dig for different types (if any): in this case you have not, so, just remove them :)